src/librustc_driver/driver.rs

```diff
@@ -216,6 +216,13 @@ pub fn compile_input(sess: &Session,
         })??
     };
 
+    if sess.opts.debugging_opts.print_type_sizes {
```

Comment:
The closure we pass into `phase_3_run_analysis_passes`?

What is the benefit you see of putting it in there? Is there some internal instrumentation that needs to be aware of this print pass taking place?